sw/device/lib/testing/dif/mock_dif_uart_test.cc

```diff
@@ -0,0 +1,228 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/mock_mmio.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+
+extern "C" {
+#include "sw/device/lib/dif/dif_uart.h"
+#include "uart_regs.h"  // Generated.
+}  // extern "C"
+
+namespace dif_uart_test {
+using ::mock_mmio::MmioTest;
+using ::testing::Test;
+
+class DifUartTestBasic : public Test, public MmioTest {
+ protected:
+  /**
+   * Sets up the default UART test configuration, the expectations are that the
+   * individual tests can further adjust the relevant values.
+   */
+  void SetUp() override {
+    // NCO is calculated as NCO = 2^20 * fbaud / fpclk, so the following values
+    // should result in NCO of 1.
+    dif_uart_config_pointer()->baudrate = 1;
+    dif_uart_config_pointer()->clk_freq_hz = 1048576;
+
+    // Initialise to defaults
+    dif_uart_config_pointer()->parity_enable = kDifUartDisable;
+    dif_uart_config_pointer()->parity = kDifUartParityEven;
```

Comment:
This configuration should be done in the definition of `dif_uart_config_`, i.e.,
```cc
 private:
  dif_uart_config_t dif_uart_config_{
    /* baudrate = */ 1,
    // ... 
  };
```

Also something something we should maybe consider allowing designated initialization in C++, since it's coming in 20 and GCC/Clang both have working extensions for it...